### PR TITLE
add example to demonstrate a rolling restart of the deployment

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
This PR add `examples/dynamic-client/deployment_rollout-restart.py` python script, to demonstrate how to perform a rolling restart on a deployment resource.

The output of the script looks like:

deployment_rolling_restart

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-client/python/issues/1378

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: examples/dynamic-client/deployment_rolling_restart.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    import datetime
+    import pytz
+
+    # Creating a dynamic client
+    client = dynamic.DynamicClient(
+        api_client.ApiClient(configuration=config.load_kube_config())
+    )
+
+    # fetching the deployment api
+    api = client.resources.get(api_version="apps/v1", kind="Deployment")
```
Comment (yliaog) [inline examples/dynamic-client/deployment_rolling_restart.py:35]: any reason for using dynamic client, instead of using the typed client?

Comment (Priyankasaggu11929) [reply examples/dynamic-client/deployment_rolling_restart.py:35]: HI @yliaog, thanks for the reveiw.

No, there is no specific reason for using `dynamic client`. I've just replicated the same example using the `typed client`. If required, I can replace this example with the one using `typed client`.

Thank you!

Comment (yliaog) [reply examples/dynamic-client/deployment_rolling_restart.py:35]: ok. i think typed client would benefit from such an example too.

Comment (Priyankasaggu11929) [reply examples/dynamic-client/deployment_rolling_restart.py:35]: Thanks @yliaog. I'll add the same example using the `typed client` too.

**Pull request merged**